torch/_inductor/codegen/wrapper.py

```diff
@@ -586,10 +587,60 @@ def __str__(self) -> str:
         return f"{type(self).__name__}({', '.join(args)})"
 
 
+class EfficientPeakEstimate:
+    def __init__(self):
+        from ..memory import estimate_peak_memory
+
+        self.overall_peak_memory, peak_by_scheduler_node = estimate_peak_memory(
+            V.graph.scheduler.nodes,
+            {},
```

Comment:
Sorry, one last thing, `names_to_freeable_bufs` is important for the backward when we need to know which activations will deallocate in order to have an accurate memory estimate